### PR TITLE
Added tests for reucers search and tasks

### PR DESCRIPTION
**Pull request by aiberry**

### CODE REVIEW

File: src/redusers/search.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import { SEARCH } from '../actions/searchFilter.js';
 
 export default function search(state = '', action) {
-    if (action.type === SEARCH) {
+    if (action && action.type === SEARCH) {
```
Comment (pavtaras) [inline src/redusers/search.js]: just `action.type === SEARCH` (action cannot be without `type` property)

File: src/redusers/search.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import { SEARCH } from '../actions/searchFilter.js';
 
 export default function search(state = '', action) {
```
Comment (pavtaras) [inline src/redusers/search.js]: `export default function searchReducer`

File: src/redusers/search.test.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    it('should handle SEARCH', () => {
+        expect(search(initStoreState.search, searchFilter('some'))).toEqual('some');
+
+        expect(search(undefined, searchFilter('some'))).toEqual('some');
+
+        expect(search(undefined, undefined)).toEqual('');
+
+        expect(search('defined', searchFilter('some'))).toEqual('some');
+
+        expect(search('defined', searchFilter(''))).toEqual('');
```
Comment (pavtaras) [inline src/redusers/search.test.js]: u just need to check 2 cases, no need to test default parameters:

```js
expect(searchReducer('oldSearch', searchFilter('newSearch')))
  .toEqual('newSearch');


expect(searchReducer('oldSearch', { type: 'another_type' }))
  .toEqual('oldSearch');
```

File: src/redusers/tasks.test.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
+/* eslint-disable no-undef */
+import { initStoreState } from '../constants';
+import addTask from '../actions/addTask.js';
+import taskClicked from '../actions/taskClicked.js';
+import tasks from './tasks.js';
+
+describe('todos reducer, tasks part', () => {
+    let store;
+    beforeEach(() => (store = initStoreState));
```
Comment (pavtaras) [inline src/redusers/tasks.test.js]: it is `state`

File: src/redusers/tasks.test.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+    it('should handle TASK_CLICKED', () => {
```
Comment (pavtaras) [inline src/redusers/tasks.test.js]:
```js
it('marks task as done if it is undone', () => {
  // ...
})

it('marks task as undone if it is done', () => {
  // ...
})
```

File: src/redusers/tasks.test.js [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+        expect(tasks(store.tasks, {})).toHaveProperty('0.status', 'undone');
+
+        expect(tasks(store.tasks, taskClicked(store.tasks[0]))).toHaveProperty(
+            '0.status',
+            'done'
+        );
+    });
+
+    it('should handle ADD_TASK', () => {
```
Comment (pavtaras) [inline src/redusers/tasks.test.js]: it('should add task to tasks list', ...)

File: src/redusers/tasks.test.js [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+        // expect(reducer(undefined, {})).toEqual(store)
+        // expect(reducer(undefined, {})).toHaveProperty('search', '')
+        // expect(reducer(undefined, {})).toHaveProperty('tasks')
+
+        expect(tasks(undefined, {})).toEqual(store.tasks);
+
+        let singleTaskAction = addTask('testText');
+        let expectedTaskList = [...initStoreState.tasks, singleTaskAction.payload];
+        expect(tasks(undefined, singleTaskAction)).toEqual(expectedTaskList);
+    });
```
Comment (pavtaras) [inline src/redusers/tasks.test.js]: no need

File: src/redusers/search.js (hunk 2 above)
Comment (aiberry) [reply src/redusers/search.js]: Is the same with tasks -> tasksReducer?

**Pull request merged**